### DISCUSSION TITLE
Trigger a workflow on change to the yml file itself

### DISCUSSION THREAD

**Discussion by kichalla** [community/community#26687] [Actions]
<p>I have 6 yml based workflows under my <code>.github/workflows</code> directory. I want these workflows to be triggered only when content under certain folder is updated but also when the particular yml file is updated too. Following is how I had defined for this purpose.</p>
<pre><code>on:
  push:
    paths:
      - sampleApps/dotNetCore/**
    branches:
      - master
      - actions/**
</code></pre>
<p><strong>Expected: </strong> A build would be kicked off since I updated the yml file itself.</p>
<p><strong>Issue</strong> : I do not see a workflow kicking off when I pushed this new workflow yml file (note that I had not made any changes to any files under the folder mentioned in the above <code>paths</code> folder)</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/kichalla">@kichalla</a> ,</p>
<p>According to your request, you can try to configure the path filters as below examples:</p>
<p>for example have two workflows in the repository:  <strong>CI</strong> ( <strong>.github/workflows/CI.yml</strong> ) and <strong>Test</strong> ( <strong>.github/workflows/Test.yml</strong> ).</p>
<pre><code>## .github/workflows/CI.yml

name: CI
on:
  push:
    paths:
      - '**'
      - '!.github/workflows/**'
      - '.github/workflows/CI.yml'
</code></pre>
<p>The <strong>CI</strong> workflow will be triggered only when one of the below two situations occurs:</p>
<p>● The workflow yaml file " <strong>.github/workflows/CI.yml</strong>" is changed.</p>
<p>● Any file <strong>outside</strong> of the directory " <strong>.github/workflows</strong>" is changed.</p>
<p>The  <strong>Test </strong> workflow is similar:</p>
<pre><code>## .github/workflows/Test.yml

name: Test
on:
  push:
    paths:
      - '**'
      - '!.github/workflows/**'
      - '.github/workflows/Test.yml'
</code></pre>
<p><strong>Note:</strong></p>
<pre><code>&lt;font size="3"&gt;In each workflow yaml file, you need pay attention to order of the patterns, because:
● A matching negative pattern (prefixed with !) after a positive match will exclude the path.
● A matching positive pattern after a negative match will include the path again.&lt;/font&gt;
</code></pre>
<p>More details about path filters, you can reference here: <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths</a></p>

**Comment by kichalla** [comment]
<p>Thanks for the solution <a class="mention" href="/brightran">@brightran</a>. But it is kind of surprising that I have to do the suggested things to make this work. I feel this should just be the automatic behavior.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/kichalla">@kichalla</a> ,</p>
<p>By default, the push event is available to all files in the repository if no path filters are set.</p>
<p>If you have set path filters on push event, the workflow will be triggered only when at least one of the modified files matches the path filters. So if you want to set which files can (or can’t) trigger the workflow on push event, you need to use the path filters.</p>

**Reply by tanujazz3** [reply]
Can we trigger the workflow on change of multiple files so making the change for the all the files mandatory in a directory ?

**Reply by airtower-luna** [reply]
The workflow will run as long as *any* file matched by the path filter changed.

**Comment by kichalla** [comment]
<p>Ok. Thanks <a class="mention" href="/brightran">@brightran</a> </p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="alxmrtn" data-post="6" data-topic="17792">
<div class="title">
<div class="quote-controls"></div>
 alxmrtn:</div>
<blockquote>
<p><code>      - './github/workflows/cdk.yml'</code></p>
</blockquote>
</aside>
<p>The path is incorrect, the directory is called <code>.github/</code> (otherwise your workflows wouldn’t work at all). Try removing the surplus slash between the dot and <code>github</code>. :wink:</p>

**Comment by alxmrtn** [comment]
<p>I feel so unbelievably dumb :face_exhaling: can’t believe I missed that - thank you</p>
